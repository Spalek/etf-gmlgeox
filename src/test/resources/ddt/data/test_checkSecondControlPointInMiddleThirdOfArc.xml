<ii:GeometryCollection xmlns:gml="http://www.opengis.net/gml/3.2"
 xmlns:ii="http://www.interactive-instruments.de/test" xmlns:xlink="http://www.w3.org/1999/xlink"
 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
 <gml:boundedBy>
  <gml:Envelope srsName="http://www.opengis.net/def/crs/EPSG/0/4326">
   <gml:lowerCorner>0 0</gml:lowerCorner>
   <gml:upperCorner>90 90</gml:upperCorner>
  </gml:Envelope>
 </gml:boundedBy>
 <ii:member>
  <gml:CompositeCurve gml:id="CC1" srsName="urn:adv:crs:etrs89_utm33">
   <gml:curveMember>
    <gml:Curve gml:id="CC1.2">
     <gml:segments>
      <gml:Arc numArc="1">
       <gml:posList>-1 0 0 -1 0 1</gml:posList>
      </gml:Arc>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
  </gml:CompositeCurve>
 </ii:member>
 <ii:member>
  <gml:CompositeCurve gml:id="CC2" srsName="urn:adv:crs:etrs89_utm33">
   <gml:curveMember>
    <gml:Curve gml:id="CC2.2">
     <gml:segments>
      <gml:ArcString numArc="1">
       <gml:posList>-1 0 0 -1 1 0 1 1 2 0</gml:posList>
      </gml:ArcString>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
  </gml:CompositeCurve>
 </ii:member>
 <ii:member>
  <gml:CompositeCurve gml:id="CC3" srsName="urn:adv:crs:etrs89_utm33">
   <gml:curveMember>
    <gml:Curve gml:id="CC3.2">
     <gml:segments>
      <gml:ArcString numArc="1">
       <gml:posList>-10 0 -9 1 10 0</gml:posList>
      </gml:ArcString>
     </gml:segments>
    </gml:Curve>
   </gml:curveMember>
  </gml:CompositeCurve>
 </ii:member>
</ii:GeometryCollection>
