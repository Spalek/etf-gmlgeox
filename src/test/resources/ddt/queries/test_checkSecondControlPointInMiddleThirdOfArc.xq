import module namespace ggeo = 'https://modules.etf-validator.net/gmlgeox/2';

declare namespace gml = 'http://www.opengis.net/gml/3.2';
declare namespace ii = 'http://www.interactive-instruments.de/test';

declare variable $dbname external := 'GMLGEOX-JUNIT-TEST-DB-000';
let $members := db:open($dbname)//ii:member
let $init := ggeo:init($dbname)

let $geometries := $members/*[.//*[self::gml:Arc or self::gml:ArcString]]
return
 <test_checkSecondControlPointInMiddleThirdOfArc>    
   {
    for $geom in $geometries
    let $arcStrings := $geom//*[self::gml:Arc or self::gml:ArcString]
    let $invalidCase :=
     (for $arcString in $arcStrings
      return
        ggeo:checkSecondControlPointInMiddleThirdOfArc($arcString)
      )[position() le 1]
    let $result :=
      if($invalidCase) then
        'failed'
      else
        'passed'
    return 
      <test geom='{$geom/@gml:id}'>{$result}</test>    
   }
 </test_checkSecondControlPointInMiddleThirdOfArc>
