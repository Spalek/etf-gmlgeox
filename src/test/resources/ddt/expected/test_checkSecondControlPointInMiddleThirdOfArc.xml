<test_checkSecondControlPointInMiddleThirdOfArc>
  <test geom="CC1">passed</test>
  <test geom="CC2">passed</test>
  <test geom="CC3">failed</test>
</test_checkSecondControlPointInMiddleThirdOfArc>